### PR TITLE
Feature/mat pointer

### PR DESCRIPTION
**Pull request by deadprogram**
Addresses #137 by using `Mat*` params to indicate "output" params that can be changed by the OpenCV function.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,4 +26,16 @@ Here is how to contribute back some code or documentation:
...
+
+Here are a few basic guidelines on how to add a function from OpenCV to GoCV:
+
+- Please open a Github issue. We want to help, and also make sure that there is no duplications of efforts. Sometimes what you need is already being worked on by someone else.
+- Use the proper Go style naming `MissingFunction()` for the Go wrapper.
+- Make any output parameters `Mat*` to indicate to developers that the underlying OpenCV data will be changed by the function.
+- Use Go types when possible as parameters for example `image.Point` and then convert to the appropriate OpenCV struct. Also define a new type based on `int` and `const` values instead of just passing "magic numbers" as params. For example, the `VideoCaptureProperties` type used in `videoio.go`.
+- Always add the function to the GoCV file named the same as the OpenCV module to which the function belongs.
+- If the new function is in a module that is not yet implemented by GoCV, a new set of files for that module will need to be added.
+- Always add a "smoke" test for the new function being added. We are not testing OpenCV itself, but just the GoCV wrapper, so all that is needed generally is just exercising the new function.
```
Comment (willdurand) [inline CONTRIBUTING.md]: should we add a note about the `WithParams` functions?

File: imgproc.go [hunk 2]
```diff
@@ -964,7 +964,7 @@ func GetPerspectiveTransform(src, dst []image.Point) Mat {
 //
 // For further details, please see:
 // https://docs.opencv.org/3.3.1/d6/d6e/group__imgproc__draw.html#ga746c0625f1781f1ffc9056259103edbc
-func DrawContours(src Mat, contours [][]image.Point, contourIdx int, c color.RGBA, thickness int) {
+func DrawContours(src *Mat, contours [][]image.Point, contourIdx int, c color.RGBA, thickness int) {
```
Comment (willdurand) [inline imgproc.go]: I guess it is not `src` but `m` here

File: CONTRIBUTING.md (hunk 1 above)
Comment (deadprogram) [reply CONTRIBUTING.md]: Done.

File: imgproc.go (hunk 2 above)
Comment (deadprogram) [reply imgproc.go]: Renamed.

**Comment by maruel** [comment]
lgtm

**Pull request merged**